include/fastdds/rtps/common/CacheChange.h

```diff
@@ -247,9 +247,17 @@ struct RTPS_DllAPI CacheChange_t
     {
         if (fragment_size_ > 0)
         {
-            if (initial_fragment == first_missing_fragment_)
+            uint32_t last_fragment = initial_fragment + num_of_fragments;
+            if (initial_fragment <= first_missing_fragment_)
             {
-                first_missing_fragment_ += num_of_fragments;
+                // Perform first = *first until first >= last_received
+                while (first_missing_fragment_ < last_fragment)
+                {
+                    size_t offset = fragment_size_;
+                    offset *= first_missing_fragment_;
+                    uint32_t* fragment = (uint32_t*)&serializedPayload.data[offset];
```

Comment:
Try to adopt the habit to use C++11 castings.

Reply:
done